src/netbase.cpp

```diff
@@ -468,7 +468,15 @@ SOCKET CreateSocket(const CService &addrConnect)
     return hSocket;
 }
 
-bool ConnectSocketDirectly(const CService &addrConnect, const SOCKET& hSocket, int nTimeout)
+void LogConnectFailure(const std::string& error_message, bool manual_connection) {
```

Comment:
I know this is annoying, but can you use argument forwarding or make this variadic? That would avoid allocation when the message is not going to be logged.